script.js

```diff
@@ -0,0 +1,151 @@
+// Assignment Code
+var acc = 0;
+
+var passwordArray = [];
+var characters = [];
+let zeroCheck;
+var abc = ["a","b","c","d","e","f","g","h","i","j","k","l","m","n","o","p","q","r","s","t","u","v","w","x","y","z"];
+var ABC = ["A","B","C","D","E","F","G","H","I","J","K","L","M","N","O","P","Q","R","S","T","U","V","W","X","Y","Z"];
+var numbers = ["0","1","2","3","5","6","7","8","9"]
+var special = ['"',
+' ',"!",'"',"#","$","%","&","'","(",")","*","+",",","-",".","/",":",";","<","=",">","?","@","[","\\","]","^","_","`","{","|","}","~"];
+
+//let zeroCheck will be used to check instances of zero, for validation purposes!
+
+var prompt1 = prompt('Please enter a number between 8 and 128')
+// initial prompt ask
+var passwordLength = parseInt(prompt1)
+//turns prompt response to numerical form
+var vrai
+
+if (passwordLength > 7 && passwordLength < 129) {vrai = true}else{vrai = false}
+//this logic sets up a condition for the while statement
+while (vrai === false){
+  //if the initial conditions aren't met the initial prompt is repeted
+  prompt1 = prompt('Please enter a number between 8 and 128')
+  passwordLength = parseInt(prompt1);
+  if (passwordLength > 7 && passwordLength < 129) {vrai = true}
+  //once conditions for truth are met - while loop broken 
+
+  //now we are going to work on the prompts
+
+  
+
+
+}
+alert("You need to choose at least one of the following character types for your password!") 
+//this asks the question!!!!!
+
+
+
+//this adds one to the accumalator if confirmed.  and not if not.  ths will be repeated for each of the character types!
+if (confirm("Do you want to include lower case letters????")){
```

Comment:
It would probably read better if we we're to take the confirm() function result out of the if() statement condition and store it in a variable such as const lowerCaseEnabled = confirm() and then we could do if (lowerCaseEnabled)